### PR TITLE
Fixes #2865 dockerize Docusaurus

### LINKED ISSUE #2865: Dockerize Docusaurus service
In https://github.com/Seneca-CDOT/telescope/issues/2792, we added Docusaurus. We need to dockerize it, also be connected to our  Nginx API gateway for routing.

### PR DESCRIPTION
**Pull request by cindyledev**
## Issue This PR Addresses
Fixes #2865

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

As per Slack

> @TDDR: @cindyledev, no pressure, but will you be able to submit a PR today for the work you have done on https://github.com/Seneca-CDOT/telescope/pull/2942?

Note: This PR only Dockerizes Docusaurus, this does not touch any of the nginx requirements. We can file another issue to address the nginx requirement or help me get the nginx working in this PR.

## Steps to test the PR
To test this PR locally:
1. Add my repository as a remote: `git remote add cindyledev https://github.com/cindyledev/telescope.git`
2. Fetch my branches: `git fetch cindyledev`
3. Checkout my branch: `git checkout issue-2865`
4. Change directory to src/docs: `src/docs`

To run `development` layer of the docker container:
1. Build the container but only up to the `development` layer, name it `docs` and tag it as `dev`: `docker build --target development -t docs:dev .`
2. Run the container in port 4631: `docker run -p 4631:4631 docs:dev`
3. Go to `localhost:4631` in the browser
4. List all the running containers: `docker ps`
5. Note the container id
6. Stop the container: `docker container stop <container id>`

To run all the layers in the docker container:
1. Build the container, name it `docusaurus` and tag it as `latest`: `docker build -t docusaurus:latest .`
2. Run the container in port 4631: `docker run --rm -p 4631:80 docusaurus:latest`
3. Go to `localhost:4631` in the browser

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/docs/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+build/
+node_modules
```
Comment (humphd) [inline src/docs/.dockerignore:2]: nit: trailing `/` on directories with Docker

File: src/docs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+FROM node:lts as base
+
+# Reduce npm log spam and colour during install within Docker
+ENV NPM_CONFIG_LOGLEVEL=warn
+ENV NPM_CONFIG_COLOR=false
+
+# We'll run the app as the `node` user, so put it in their home directory
+WORKDIR /home/node/app
+# Install pnpm
+RUN npm install -g pnpm
```
Comment (humphd) [inline src/docs/Dockerfile:12]: We'll be able to use our base image soon here, @Kevan-Y.

**Review by humphd — APPROVED** [review]

File: src/docs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Production ##################################################################
+# Also define a production target which doesn't use devDeps
+FROM base as production
+WORKDIR /home/node/app
+COPY --chown=node:node --from=development /home/node/app/node_modules /home/node/app/node_modules
+
+## Deploy ######################################################################
+# Use a stable nginx image
+FROM nginx:stable-alpine as deploy
```
Comment (manekenpix) [inline src/docs/Dockerfile:37]: I'm curious about this: are we going to deploy another instance of `nginx` or are we going to use the one serving our main pages to serve the docs?

Comment (humphd) [reply src/docs/Dockerfile:37]: I'm not entirely sure how we'd do this with a single instance from a Docker point of view.  We have totally separate packages for `src/web` and `src/docs`.  Without merging them, how should we share the server?

I agree, having a single static host makes a lot of sense.  But we could move to that in a follow-up.

Comment (humphd) [reply src/docs/Dockerfile:37]: Discussing this with @manekenpix, in a follow-up we could do:

```
src/web/Dockerfile
src/web/app
src/web/docs
```

Now we can build both static sites (next.js, Docusaurus) and host them on the same nginx instance.

Any Docker folks want to try this one?

Comment (humphd) [reply src/docs/Dockerfile:37]: Filed https://github.com/Seneca-CDOT/telescope/issues/2993 for later.

Comment (cindyledev) [reply src/docs/Dockerfile:37]: Moving Docusaurus into /web/ makes so much sense! We should also rename the `/docs` directory as well. I realized some of the file paths currently look like `src/docs/docs/*` and that will likely be prone to errors.

I will make a comment on the follow-up issue above.

**Review by humphd — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Pull request merged**